### PR TITLE
Doc: Added docstring to Groupby mean

### PR DESCRIPTION
**Pull request by cxl923cc**
Added examples of the usage and what is returned

### CODE REVIEW

**Comment by Cheukting** [comment]
Thanks for the PR! There is some PEP 8 problem (basically formatting issues) which needed to be fixed. Also you may want to use `git diff upstream/master -u -- "*.py" | flake8 --diff` to check if there's PEP 8 problem before pushing. Also, you may want to  use "DOC: update the pandas.core.groupby.GroupBy.mean docstring" as the PR title so developers knows exactly which method/functions you are updating the docstring.

File: pandas/core/groupby/groupby.py [hunk 1]
```diff
@@ -1266,8 +1266,40 @@ def count(self):
...
+        
+        Example of groupby one column:
+        ------------------------------
+        >>> df = pd.DataFrame({'A': [1, 1, 2, 1, 2],
+                               'B': [np.nan, 2, 3, 4, 5]}, columns=['A', 'B'])
+        >>> g = df.groupby('A')['B'].mean()
+        >>> g
+        A
+        1    3.0
+        2    4.0
```
Comment (Cheukting) [inline pandas/core/groupby/groupby.py]: You may want to put the Example at the end, see https://python-sprints.github.io/pandas/guide/pandas_docstring.html

File: pandas/core/groupby/groupby.py [hunk 2]
```diff
@@ -1266,8 +1266,40 @@ def count(self):
...
+                               'B': [np.nan, 2, 3, 4, 5],
+                               'C': [1, 2, 1, 1, 2]}, columns=['A', 'B', 'C'])
+        >>> g = df.groupby(['A', 'C'])['B'].mean()
+        >>> g
+        A  C
+        1  1    4.0
+           2    2.0
+        2  1    3.0
+           2    5.0
+
```
Comment (Cheukting) [inline pandas/core/groupby/groupby.py]: You may want to put the Example at the end, see https://python-sprints.github.io/pandas/guide/pandas_docstring.html

File: pandas/core/groupby/groupby.py [hunk 3]
```diff
@@ -1266,8 +1266,40 @@ def count(self):
     def mean(self, *args, **kwargs):
         """
         Compute mean of groups, excluding missing values
+
```
Comment (Cheukting) [inline pandas/core/groupby/groupby.py]: May be you can try adding long summary and Parameters as well, see https://python-sprints.github.io/pandas/guide/pandas_docstring.html

**Review by Cheukting — CHANGES_REQUESTED** [review]

**Comment by cxl923cc** [comment]
This is my first pull request. Could someone please help to review the new commit I have made and let me know whether I should make any further change to the Docstring?

File: pandas/core/groupby/groupby.py [hunk 4]
```diff
@@ -1265,9 +1265,40 @@ def count(self):
     @Appender(_doc_template)
     def mean(self, *args, **kwargs):
         """
-        Compute mean of groups, excluding missing values
+        Compute mean of the target column for groups that are defined by the groupby columns.
+        Missing values are excluded in computing the mean.
```
Comment (jreback) [inline pandas/core/groupby/groupby.py]: blank line here for the Extended Summary

File: pandas/core/groupby/groupby.py [hunk 5]
```diff
@@ -1265,9 +1265,40 @@ def count(self):
...
+        If there is one groupby column, the groups are the unique values in the specified groupby column.
+        If there are multiple groupby columns, the groups are the unique combinations of the specified groupby columns.
+
+        Returns
+        -------
+        pandas.core.series.Series
+            The average of the target column ('B' in the examples below)
+            grouped by the groupby columns ('A' and ['A', 'C'] in the examples below)
+
+        Example of groupby one column:the groups are stored as index in the result.
```
Comment (jreback) [inline pandas/core/groupby/groupby.py]: don't use underlines, rather use a blank line

Comment (jreback) [inline pandas/core/groupby/groupby.py]: make this a sentence.
Groubpy by one column. The result index are the group labels.

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @cxl923cc and @mroeschke for the fixup!

**Pull request merged**